src/WebJobs.Extensions.DurableTask/LinuxAppServiceFileLogger.cs

```diff
@@ -111,6 +111,8 @@ private async Task ProcessLogQueue(object state)
                 await Task.Delay(TimeSpan.FromSeconds(this.FlushFrequencySeconds), this.cancellationTokenSource.Token).ContinueWith(task => { });
             }
 
+            await this.InternalProcessLogQueue();
```

Comment:
I'm not sure if it is best practices to await after a cancellation token is cancelled. @cgillum, thoughts?

Reply:
To clarify, I think it makes sense to, I'm just not sure what the expectations are in .NET when a token is cancelled.

Reply:
Got it. I see how this could be controversial.  Note that this gets called in here:

```C#
        /// <summary>
        /// Flushes the write buffer, stops writing to logfile afterwards.
        /// </summary>
        /// <param name="timeSpan">Timeout in milliseconds for flushing task.</param>
        public void Stop(TimeSpan timeSpan)
        {
            this.cancellationTokenSource.Cancel(); // this is not awaited

            try
            {
                this.outputTask?.Wait(timeSpan);
            }
            catch (Exception)
            {
                // ignored
            }
        }
```

which in turn is called in the dispose method of DurableTaskExtension.cs

```C#
        public void Dispose()
        {
            // Not flushing the linux logger may lead to lost logs
            // 40 seconds timeout because we write normally every 30 seconds, so we're just
            // adding an extra 10 seconds to flush.
            LinuxAppServiceLogger.Logger?.Stop(TimeSpan.FromSeconds(40));
            this.HttpApiHandler?.Dispose();
            this.eventSourceListener?.Dispose();
        }
```

Reply:
Without the highlighted change though, cancelling a token while the buffer still has contents can result is missed logs. I've also already notified the azure-functions-host folks about this potential problem

Reply:
I think of CancellationToken as just a tool for delivering notifications. Any best practices as far as what to do with that notification would be specific to the application. In this case, I agree we should flush out our log buffer. You just may need to be careful about potential race conditions. For example, the listener could be disposed at the same time that we're flushing the logs. We just need to be careful that this won't cause any problems for us.

Reply:
I'm giving the flush operation _at minimum_ 10 seconds to flush, and 40 seconds in the best case scenario. I believe that _should_ be enough if we assume the app is otherwise in a healthy state.